extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java

```diff
@@ -383,7 +384,7 @@ private Type verifyQueryResultType(Type t) {
                 }
                 return t;
             }
-        } else if (!DotNames.OBJECT.equals(t.name())) {
+        } else if (!DotNames.OBJECT.equals(t.name()) && !DotNames.STRING.equals(t.name()) && !isIntLongOrBoolean(t.name())) {
```

Comment:
Minor gripe: if I understand the logic of this method right, I think you already handle the `isIntLongOrBoolean` case at the beginning of the method. And I think we should better handle all these special cases there.

Reply:
You are absolutely correct! I'll fix it